### PR TITLE
workflows: simply workflow with a matrix

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Use a matrix to determine base distros for each job and use that to select a base container image.

### CODE REVIEW

File: sambacc/smbconf_api.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        self._data[name] = value
+
+    def __iter__(self) -> typing.Iterator[str]:
+        return iter(self._data.keys())
+
+    def import_smbconf(
+        self, src: ConfigStore, batch_size: typing.Optional[int] = None
+    ) -> None:
+        """Import content from one SMBConf configuration object into the
+        current SMBConf confiugration object.
```
Comment (scara) [inline sambacc/smbconf_api.py:55]: Trivial typo here: `confiugration` => `configuration`

HTH,
Matteo

File: sambacc/smbconf_samba.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        self._smbconf.create_set_share(name, value)
+
+    def __iter__(self) -> typing.Iterator[str]:
+        return iter(self._smbconf.share_names())
+
+    def import_smbconf(
+        self, src: ConfigStore, batch_size: typing.Optional[int] = 100
+    ) -> None:
+        """Import content from one SMBConf configuration object into the
+        current SMBConf confiugration object.
```
Comment (scara) [inline sambacc/smbconf_samba.py:120]: Trivial typo here: `confiugration` => `configuration`

HTH,
Matteo

File: sambacc/smbconf_api.py (hunk 1 above)
Comment (phlogistonjohn) [reply sambacc/smbconf_api.py:55]: Thank you. I have fixed the mistake in PR #85 - this branch is based on the branch for PR #85 as this one was me experimenting with changes to the CI. I was being lazy when I filed it and did not cleanly separate the streams of work. When I next rebase the changes related to global config on AD DCs should disappear from this PR.

**Comment by phlogistonjohn** [comment]
NOTE: This will need corresponding updates to mergify config and branch protection rules in the github ui. I'll update the mergify config in another PR and update the GUI manually after merge.

**Comment by anoopcs9** [comment]
@Mergifyio rebase

**Review by anoopcs9 — APPROVED** [review]
lgtm.

**Comment by anoopcs9** [comment]
@Mergifyio rebase

**Pull request merged**